### PR TITLE
BAEL-5301 Retrying Feign Calls

### PR DESCRIPTION
**Pull request by mdabrowski-eu**
http://jira.baeldung.com/browse/BAEL-5301
https://drafts.baeldung.com/wp-admin/post.php?post=125475&action=edit

### CODE REVIEW

File: feign/src/main/java/com/baeldung/feign/retry/Custom5xxErrorDecoder.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import static feign.FeignException.errorStatus;
+
+public class Custom5xxErrorDecoder implements ErrorDecoder {
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        FeignException exception = errorStatus(methodKey, response);
+        int status = response.status();
+        if (status >= 500) {
+            return new RetryableException(
+                    response.status(),
```
Comment (doodoroma) [inline feign/src/main/java/com/baeldung/feign/retry/Custom5xxErrorDecoder.java:18]: Line continuation needs 2-space indentation

File: feign/src/main/java/com/baeldung/feign/retry/ResilientFeignClientBuilder.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import java.util.concurrent.TimeUnit;
+
+@Getter
+public class ResilientFeignClientBuilder {
+    public BookClient bookClient = createClient(BookClient.class, "http://localhost:8081/api/books");
+
+    public static <T> T createClient(Class<T> type, String uri) {
+        return Feign.builder()
+            .client(new OkHttpClient())
```
Comment (doodoroma) [inline feign/src/main/java/com/baeldung/feign/retry/ResilientFeignClientBuilder.java:21]: Here as well

File: feign/src/test/java/com/baeldung/feign/retry/Custom5xxErrorDecoderUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        // then
+        assertTrue(exception instanceof FeignException);
+        assertFalse(exception instanceof RetryableException);
+    }
+
+    @NotNull
+    private Response responseStub(int status) {
+        return Response.builder()
+                .request(Request.create(
```
Comment (doodoroma) [inline feign/src/test/java/com/baeldung/feign/retry/Custom5xxErrorDecoderUnitTest.java:46]: Here as well

File: feign/src/main/java/com/baeldung/feign/retry/Custom5xxErrorDecoder.java (hunk 1 above)
Comment (mdabrowski-eu) [reply feign/src/main/java/com/baeldung/feign/retry/Custom5xxErrorDecoder.java]: Done.

File: feign/src/test/java/com/baeldung/feign/retry/Custom5xxErrorDecoderUnitTest.java (hunk 3 above)
Comment (mdabrowski-eu) [reply feign/src/test/java/com/baeldung/feign/retry/Custom5xxErrorDecoderUnitTest.java]: Done.

File: feign/src/main/java/com/baeldung/feign/retry/ResilientFeignClientBuilder.java (hunk 2 above)
Comment (mdabrowski-eu) [reply feign/src/main/java/com/baeldung/feign/retry/ResilientFeignClientBuilder.java]: Done.

**Pull request merged**